statsmodels/base/model.py

```diff
@@ -751,7 +756,21 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
                 exog = exog[:, None]
             exog = np.atleast_2d(exog)  # needed in count model shape[1]
 
-        return self.model.predict(self.params, exog, *args, **kwargs)
+        predict_results = self.model.predict(self.params, exog, *args, **kwargs)
+
+        if (isinstance(orig_exog, pd.Series) or isinstance(orig_exog, pd.DataFrame) and
```

Comment:
why did you switch to an instance check instead of just checking for an attribute `index`?